sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -141,7 +139,23 @@ public Symbol generateQuerySymbol(Optional<Expression> whereExpression, Expressi
     }
 
     private FunctionInfo getFunctionInfo(String name, List<DataType> argumentTypes) {
-        return functions.getSafe(name, argumentTypes).info();
+        FunctionImplementation implementation = null;
+        String exceptionMessage = null;
+        try {
+            implementation = functions.getBuiltin(name, argumentTypes);
+        } catch (IllegalArgumentException e) {
+            if (e.getMessage() != null && !e.getMessage().isEmpty()) {
+                exceptionMessage = e.getMessage();
```

Comment:
Would it be possible to keep the "format unknown function error" functionality within the Functions class? Also, without the context that there is going to be a UDF fallback this change isolated looks a bit odd. Maybe it's better to keep this in the feature branch for now